code/game/objects/items/devices/PDA/PDA.dm

```diff
@@ -704,17 +706,15 @@ var/global/list/obj/item/device/pda/PDAs = list()
 		return
 
 	if (usr.canUseTopic(src))
-		var/obj/item/weapon/pen/O = locate() in src
-		if(O)
-			if (istype(loc, /mob))
+		if(inserted_item)
+			if(ismob(loc))
 				var/mob/M = loc
-				if(M.get_active_hand() == null)
-					M.put_in_hands(O)
-					usr << "<span class='notice'>You remove \the [O] from \the [src].</span>"
-					return
-			O.loc = get_turf(src)
-		else
-			usr << "<span class='warning'>This PDA does not have a pen in it!</span>"
+				M.put_in_hands(inserted_item)
+				usr << "<span class='notice'>You remove \the [inserted_item] from \the [src].</span>"
+				inserted_item = null
+				return
+			inserted_item.forceMove(loc)

```

Comment:
You're moving the item out of the pda but not clearing the inserter_item reference. ALso the message below will be displayed even though there was an item in the pda.